mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala

```diff
@@ -1100,14 +1100,45 @@ class LogisticRegressionModel private[spark] (
 
   private lazy val _intercept = interceptVector.toArray.head
 
+  private lazy val _interceptVector = interceptVector.toDense
+
+  private var _threshold = Double.NaN
+
+  private var _rawThreshold = Double.NaN
+
+  {
+    updateThreshold()
+  }
+
+  private def updateThreshold(): Unit = {
+    if (!isMultinomial) {
```

Comment:
Does anything need to happen in the multinomial case?

Reply:
@zhengruifeng We can only improve the performance for binary case, right?

Reply:
@srowen @huaxingao this change is only for binary classification. I found there is not similar place to change for the multinomial case.

Reply:
for multinomial case, it seems should be modified in `ProbabilisticClassifier`.

I will study whether similar change in  `ProbabilisticClassifier` can help improve performance in all `ProbabilisticClassificationModel` including multinomial logistic regression.